### PR TITLE
Added iso layer support for the GMMK Pro iso version

### PR DESCRIPTION
**Pull request by tipok**
## Description

Extended GMMK Pro keyboard with an ISO layout, which is working with the ISO version of the keyboard.
The new layer can be used for new key mappings.
Added an example mapping for iso.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* [Feature Request] GMMK Pro ISO support #13524

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/keymaps/iso/keymap.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+//      Sh_L     /        Z        X        C        V        B        N        M        ,        .        ?                 Sh_R     Up       End
+//      Ct_L     Win_L    Alt_L                               SPACE                               Alt_R    FN       Ct_R     Left     Down     Right
+
+
+    // The FN key by default maps to a momentary toggle to layer 1 to provide access to the RESET key (to put the board into bootloader mode). Without
+    // this mapping, you have to open the case to hit the button on the bottom of the PCB (near the USB cable attachment) while plugging in the USB
+    // cable to get the board into bootloader mode - definitely not fun when you're working on your QMK builds. Remove this and put it back to KC_RGUI
+    // if that's your preference.
+    //
+    // To put the keyboard in bootloader mode, use FN+backslash. If you accidentally put it into bootloader, you can just unplug the USB cable and
```
Comment (Hatbuster) [inline keyboards/gmmk/pro/keymaps/iso/keymap.c:34]: It's not FN+Backslash anymore now. Please change the description to FN+Enter or change both to another combination.

File: keyboards/gmmk/pro/keymaps/iso/keymap.c [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    // it'll be back to normal when you plug it back in.
+    [0] = LAYOUT_iso(
+        KC_ESC,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_PSCR,          KC_MUTE,
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,          KC_DEL,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_ENT,           KC_PGUP,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS,                   KC_PGDN,
```
Comment (fauxpark) [inline keyboards/gmmk/pro/keymaps/iso/keymap.c:40]:
```suggestion
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,                   KC_PGUP,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,           KC_PGDN,
```

File: keyboards/gmmk/pro/keymaps/iso/keymap.c [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH,          KC_RSFT, KC_UP,   KC_END,
+        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(1),   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT_iso(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET,            _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
```
Comment (fauxpark) [inline keyboards/gmmk/pro/keymaps/iso/keymap.c:49]:
```suggestion
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                   _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, RESET,            _______,
```

File: keyboards/gmmk/pro/pro.h [hunk 4]
```diff
@@ -48,3 +48,32 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+//  Tab  Q    W    E    R    T    Y    U    I    O    P    [    ]    Enter    PgUp
+//  Caps A    S    D    F    G    H    J    K    L    ;    "    #             PgDn
+//  Sh_L /    Z    X    C    V    B    N    M    ,    .    ?  Sh_R       Up   End
+//  Ct_L    Win_L   Alt_L      SPACE          Alt_R   FN   Ct_R     Left Down Right
+
+#define LAYOUT_iso( \
+    k13, k26, k36, k31, k33, k07, k63, k71, k76, ka6, ka7, ka3, ka5, k97,     k01, \
+    k16, k17, k27, k37, k47, k46, k56, k57, k67, k77, k87, k86, k66, ka1,     k65, \
+    k11, k10, k20, k30, k40, k41, k51, k50, k60, k70, k80, k81, k61, ka4,     k15, \
+    k21, k12, k22, k32, k42, k43, k53, k52, k62, k72, k82, k83, k84,          k25, \
```
Comment (fauxpark) [inline keyboards/gmmk/pro/pro.h:63]:
```suggestion
    k11, k10, k20, k30, k40, k41, k51, k50, k60, k70, k80, k81, k61,          k15, \
    k21, k12, k22, k32, k42, k43, k53, k52, k62, k72, k82, k83, k84, ka4,     k25, \
```
ISO Enter should be on the next line by convention:
https://github.com/qmk/qmk_firmware/tree/master/layouts/default

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/gmmk/pro/keymaps/iso/keymap.c (hunk 1 above)
Comment (fauxpark) [reply keyboards/gmmk/pro/keymaps/iso/keymap.c:34]: Fn+Backspace is a good alternative that would work for both.

**Comment by fauxpark** [comment]
You'll also need to add this to info.json.

File: keyboards/gmmk/pro/keymaps/iso/keymap.c (hunk 1 above)
Comment (Hatbuster) [reply keyboards/gmmk/pro/keymaps/iso/keymap.c:34]: Agreed, I use Fn+Backspace on my personal layout, too.

**Comment by tipok** [comment]
Thank you for the feedback, will be adding the changes later today. Regarding the info.json, do you have a tip how to fill it instead of calculating positions manually?

**Comment by fauxpark** [comment]
It will be mostly the same as the existing layout, you can use the configurator to test it by pressing Ctrl+Shift+I, and you can check the ordering is correct by turning on fast input.

**Comment by Hatbuster** [comment]
> Thank you for the feedback, will be adding the changes later today. Regarding the info.json, do you have a tip how to fill it instead of calculating positions manually?

I did most of the work for that already by hand. In an effort to not let it go to waste, here's my draft.
[info.zip](https://github.com/qmk/qmk_firmware/files/6809238/info.zip)

One or two keys might be scuffed, so do run it through the configurator to check like fauxpark suggested. I did add labels for every key, however, as appears to be better practice which the original did not follow.

**Comment by tipok** [comment]
> > Thank you for the feedback, will be adding the changes later today. Regarding the info.json, do you have a tip how to fill it instead of calculating positions manually?
> 
> I did most of the work for that already by hand. In an effort to not let it go to waste, here's my draft.
> [info.zip](https://github.com/qmk/qmk_firmware/files/6809238/info.zip)
> 
> One or two keys might be scuffed, so do run it through the configurator to check like fauxpark suggested. I did add labels for every key, however, as appears to be better practice which the original did not follow.

Thank you very much, will give it a try.

**Comment by tipok** [comment]
I've update the layout corresponding to the comments and added the layout to info.json

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**